x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/categorization_field/categorization_field.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useContext, useEffect, useState } from 'react';
+
+import { CategorizationFieldSelect } from './categorization_field_select';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { newJobCapsService } from '../../../../../../../services/new_job_capabilities_service';
+import {
+  MultiMetricJobCreator,
+  isMultiMetricJobCreator,
+  PopulationJobCreator,
+  isPopulationJobCreator,
+  AdvancedJobCreator,
+  isAdvancedJobCreator,
+} from '../../../../../common/job_creator';
+import { Description } from './description';
+
+export const CategorizationField: FC = () => {
+  const { jobCreator: jc, jobCreatorUpdate, jobCreatorUpdated } = useContext(JobCreatorContext);
+  if (
```

Comment:
AFAIK we must not return before all hooks are called in a component, I wonder if you get console warnings once this component updates? Maybe it can be solved without refactored this whole component by just wrapping it in a `WithJobCreator` component. You would do the check and return `null` in the wrapper and only render this component with `jobCreator` being always available and calling all hooks.